app/views/shared/_error_messages.html.erb

```diff
@@ -0,0 +1,12 @@
+<% if @user.errors.any? %>
+  <div id="error_explanation">
+    <div class="alert alert-danger">
+       <%t("shared.error_messages")%> <%= pluralize(@user.errors.count, "error") %>.
```

Comment:
1. sau <% <%= và trước %> luôn để 1 space a nhé
2. Chỗ này a dùng count. Ngoài ra còn có thể dùng hàm gì để lấy số lượng records k a? A so sánh các loại đó và chọn 1 loại tối ưu nhất để dùng nhé. Có 3 loại a

Reply:
Có 3 cách để đếm số lượng records
1. COUNT: không được lưu trữ trong vòng đời của đối tượng, vì vậy mỗi lần gọi phương thức này, nó sẽ thực hiện lại câu truy vấn. Tuy nhiên, truy vấn COUNT(*) trong SQL thực hiện rất nhanh
2.LENGTH: Đếm số lượng records mà không cần thực hiện thêm câu truy vấn nếu như những records đó đã được load
Ngược lại, nó sẽ load toàn bộ records vào trong bộ nhớ và thực hiện đếm(:/)
3.SIZE: Kết hợp khả năng của cả 2 phương thức trên
Nếu những records đã được load, nó sẽ thực hiện đếm mà không truy vấn vào CSDL nữa
Nếu những records chưa được load, nó sẽ thực hiện đếm sử dụng câu lệnh SQL (SELECT COUNT(*) FROM...)
=> Thằng SIZE là  bá nhất :))